ARROW-12703: [JS] Separate Table from DataFrame

domoritz opened the pull request:
Changes:

* **Breaking:** Tables are not DataFrames anymore. You can construct `DataFrame`s just like `Table`s.

TheNeuralBit reviewed (commented):
Seems like a reasonable change, I'm sure many users just want the ability to use Table/Vector without pulling in the compute code.

We should be careful since this is a breaking change. A couple of things to do:
- Make sure documentation indicates how to create a DataFrame from a Table, and that we don't have any example using DataFrame ops from a Table.
- Make sure this change is noted in release notes (in Beam we have a CHANGES.md file to help keep track of things like this. I don't think Arrow has anything similar though. I guess we'll just need to remember to update the 5.0 blog post?)

domoritz commented:
@TheNeuralBit Thank you for the review. I added documentation for constructing DataFrames from Tables and added a change note to this pull request. For 4.0, I went through all PRs. I think we can add these change notes to all PRs and then copy them into the release post. I'm happy to lead the charge for the 5.0 release notes.

domoritz commented:
Test failure is due to https://github.com/apache/arrow/pull/10314.
